packages/kbn-generate-csv/src/generate_csv.ts

```diff
@@ -256,11 +255,21 @@ export class CsvGenerator {
     let cursor: SearchCursor;
     if (this.job.pagingStrategy === 'scroll') {
       // Optional strategy: scan-and-scroll
-      cursor = new SearchCursorScroll(indexPatternTitle, settings, this.clients, this.logger);
+      cursor = new SearchCursorScroll(
+        indexPatternTitle,
+        { ...settings, taskInstanceFields: this.taskInstanceFields },
```

Comment:
I'd like to some parts more straightforward by not having to merge the taskInstance fields with settings, and not have to merge the taskInstanceFields into the CsvExportSettings interface for the SearchCursor instances. This diff outlines my idea:

```diff
diff --git packages/kbn-generate-csv/src/generate_csv.ts packages/kbn-generate-csv/src/generate_csv.ts
index 213fd902bb0..675b9e22421 100644
--- packages/kbn-generate-csv/src/generate_csv.ts
+++ packages/kbn-generate-csv/src/generate_csv.ts
@@ -224,7 +224,13 @@ export class CsvGenerator {
   public async generateData(): Promise<TaskRunResult> {
     const logger = this.logger;
     const [settings, searchSource] = await Promise.all([
-      getExportSettings(this.clients.uiSettings, this.config, this.job.browserTimezone, logger),
+      getExportSettings(
+        this.clients.uiSettings,
+        this.taskInstanceFields,
+        this.config,
+        this.job.browserTimezone,
+        logger
+      ),
       this.dependencies.searchSourceStart.create(this.job.searchSource),
     ]);
 
@@ -255,21 +261,11 @@ export class CsvGenerator {
     let cursor: SearchCursor;
     if (this.job.pagingStrategy === 'scroll') {
       // Optional strategy: scan-and-scroll
-      cursor = new SearchCursorScroll(
-        indexPatternTitle,
-        { ...settings, taskInstanceFields: this.taskInstanceFields },
-        this.clients,
-        this.logger
-      );
+      cursor = new SearchCursorScroll(indexPatternTitle, settings, this.clients, this.logger);
       logger.debug('Using search strategy: scroll');
     } else {
       // Default strategy: point-in-time
-      cursor = new SearchCursorPit(
-        indexPatternTitle,
-        { ...settings, taskInstanceFields: this.taskInstanceFields },
-        this.clients,
-        this.logger
-      );
+      cursor = new SearchCursorPit(indexPatternTitle, settings, this.clients, this.logger);
       logger.debug('Using search strategy: pit');
     }
     await cursor.initialize();
diff --git packages/kbn-generate-csv/src/generate_csv_esql.ts packages/kbn-generate-csv/src/generate_csv_esql.ts
index 20802a3c4fa..cea0838460a 100644
--- packages/kbn-generate-csv/src/generate_csv_esql.ts
+++ packages/kbn-generate-csv/src/generate_csv_esql.ts
@@ -69,6 +69,7 @@ export class CsvESQLGenerator {
   public async generateData(): Promise<TaskRunResult> {
     const settings = await getExportSettings(
       this.clients.uiSettings,
+      this.taskInstanceFields,
       this.config,
       this.job.browserTimezone,
       this.logger
diff --git packages/kbn-generate-csv/src/lib/get_export_settings.ts packages/kbn-generate-csv/src/lib/get_export_settings.ts
index 1ee32ee732e..d174b9f79b6 100644
--- packages/kbn-generate-csv/src/lib/get_export_settings.ts
+++ packages/kbn-generate-csv/src/lib/get_export_settings.ts
@@ -22,6 +22,7 @@ import { CsvPagingStrategy } from '../../types';
 
 export interface CsvExportSettings {
   timezone: string;
+  taskInstanceFields: TaskInstanceFields;
   scroll: {
     strategy?: CsvPagingStrategy;
     size: number;
@@ -42,6 +43,7 @@ export interface CsvExportSettings {
 
 export const getExportSettings = async (
   client: IUiSettingsClient,
+  taskInstanceFields: TaskInstanceFields,
   config: ReportingConfigType['csv'],
   timezone: string | undefined,
   logger: Logger
@@ -78,6 +80,7 @@ export const getExportSettings = async (
 
   return {
     timezone: setTimezone,
+    taskInstanceFields,
     scroll: {
       strategy: config.scroll.strategy as CsvPagingStrategy,
       size: config.scroll.size,
diff --git packages/kbn-generate-csv/src/lib/search_cursor.ts packages/kbn-generate-csv/src/lib/search_cursor.ts
index 8ae63d93bda..088a929030c 100644
--- packages/kbn-generate-csv/src/lib/search_cursor.ts
+++ packages/kbn-generate-csv/src/lib/search_cursor.ts
@@ -14,7 +14,6 @@ import type {
   ISearchClient,
   ISearchSource,
 } from '@kbn/data-plugin/common';
-import type { TaskInstanceFields } from '@kbn/reporting-common/types';
 import type { CsvExportSettings } from './get_export_settings';
 
 export interface SearchCursorClients {
@@ -24,10 +23,8 @@ export interface SearchCursorClients {
 
 export type SearchCursorSettings = Pick<
   CsvExportSettings,
-  'scroll' | 'includeFrozen' | 'maxConcurrentShardRequests'
-> & {
-  taskInstanceFields: TaskInstanceFields;
-};
+  'scroll' | 'includeFrozen' | 'maxConcurrentShardRequests' | 'taskInstanceFields'
+>;
 
 export abstract class SearchCursor {
   protected cursorId: string | undefined;
```

Reply:
Thanks for the pointer, I like this approach to consolidate taskInstanceFields into `CsvExportSettings`.